hnn_core/network.py

```diff
@@ -832,6 +903,320 @@ def _get_src_type_and_pos(self, gid):
+        connectivity = self._all_to_all_connect(
+            src_types, src_gids, target_types, target_gids,
+            loc, receptor, nc_dict)
+        self.connectivity_list.extend(connectivity)
+
+        src_cell = 'L2Pyr'
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        key = f'gbar_L2Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        connectivity = self._all_to_all_connect(
+            src_types, src_gids, target_types, target_gids,
+            loc, receptor, nc_dict)
+        self.connectivity_list.extend(connectivity)
+
+    def _all_to_all_connect(self, src_types, src_gids, target_types,
+                            target_gids, loc, receptor, nc_dict,
+                            allow_autapses=True, unique=False):
+        """Generate connectivity list given lists of sources and targets.
+
+        Parameters
+        ----------
+        src_types : list of str
+            List of string names of sources.
+        src_gids : list on int
+            List of integer identifiers of sources.
+        target_types : list of str
+            List of string names of targets.
+        target_gids : list on int
+            List of integer identifiers of targets.
+        loc : str
+            If 'proximal' or 'distal', the corresponding
+            dendritic sections from Cell.sect_loc['proximal']
+            or Cell.Sect_loc['distal'] are used
+        receptor : str
+            The receptor.
+        nc_dict : dict
+            The connection dictionary containing keys
+            A_delay, A_weight, lamtha, and threshold.
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        unique : bool
+            If True, each target cell gets one "unique" feed.
+            If False, all src_type cells are connected to
+            all target_type cells.
+        """
+        connectivity = list()
+        src_start = src_gids[0]  # Necessary for unique feeds
+        for target_type, target_gid in zip(target_types, target_gids):
+            if unique:
+                src_gids = [target_gid + src_start]
+                src_types = [src_types[0]]  # Assumes that all src_types match
+            for src_type, src_gid in zip(src_types, src_gids):
+                conn = {'src_type': src_type, 'src_gid': src_gid,
+                        'target_type': target_type, 'target_gid': target_gid,
+                        'loc': loc, 'receptor': receptor,
+                        'nc_dict': deepcopy(nc_dict)}
+                if not allow_autapses and src_gid == target_gid:
+                    continue
+
+                connectivity.append(conn)
+        return connectivity
+
+    def add_connection(self, connectivity):
+        """Appends connections to connectivity list
+
+        Parameters
+        ----------
+        connectivity : dict | list of dict
```

Comment:
honestly this dict is too complicated. Can't we simplify it?

Reply:
Realizing a few easy targets. We can just get rid of `src_type` and `gid_type` since there really isn't much utility in custom connection names the user will never see.

Additionally the `nc_dict` doesn't need to be nested, it's items can be packaged under the hood.